src/app/games/tafl/TaflRules.ts

```diff
@@ -0,0 +1,462 @@
+        }
+        if (back === RelativePlayer.PLAYER) {
+            return this.captureKingWithAtLeastASandwhich(state, kingCoord, left, leftCoord, right, rightCoord);
+        }
+        return null;
+    }
+    public getSurroundings(c: Coord, d: Orthogonal, player: Player, board: Table<TaflPawn>)
+    : { backCoord: Coord, back: RelativePlayer, backInRange: boolean,
+        leftCoord: Coord, left: RelativePlayer,
+        rightCoord: Coord, right: RelativePlayer }
+    {
+        const backCoord: Coord = c.getNext(d); // the piece that just move came from the front direction (by definition)
+        const backInRange: boolean = backCoord.isInRange(this.config.WIDTH, this.config.WIDTH);
+        const back: RelativePlayer = backInRange ?
+            this.getRelativeOwner(player, backCoord, board) :
+            RelativePlayer.NONE;
+
+        const leftCoord: Coord = c.getLeft(d);
+        const leftInRange: boolean = leftCoord.isInRange(this.config.WIDTH, this.config.WIDTH);
+        const left: RelativePlayer = leftInRange ?
+            this.getRelativeOwner(player, leftCoord, board) :
+            RelativePlayer.NONE;
+
+        const rightCoord: Coord = c.getRight(d);
+        const rightInRange: boolean = rightCoord.isInRange(this.config.WIDTH, this.config.WIDTH);
+        const right: RelativePlayer = rightInRange ?
+            this.getRelativeOwner(player, rightCoord, board) :
+            RelativePlayer.NONE;
+        return {
+            backCoord, back, backInRange,
+            leftCoord, left,
+            rightCoord, right,
+        };
+    }
+    private captureKingAgainstTheWall(left: RelativePlayer,
+                                      leftCoord: Coord,
+                                      right: RelativePlayer,
+                                      rightCoord: Coord,
+                                      kingCoord: Coord,
+                                      state: S)
+    : Coord
+    {
+        const LOCAL_VERBOSE: boolean = false;
+        let nbInvaders: number = (left === RelativePlayer.PLAYER ? 1 : 0);
+        nbInvaders += (right === RelativePlayer.PLAYER ? 1 : 0);
+        if (nbInvaders === 2 && this.config.BORDER_CAN_SURROUND_KING) { // 2
+            // king captured by 3 invaders against 1 border
+            display(TaflRules.VERBOSE || LOCAL_VERBOSE, 'king captured by 3 invaders against 1 border');
+            return kingCoord;
+        }
+        // those were the only two way to capture against the border
+        return null;
+    }
+    private captureKingAgainstThrone(backCoord: Coord,
+                                     kingCoord: Coord,
+                                     left: RelativePlayer,
+                                     right: RelativePlayer)
+    : Coord
+    {
+        if (this.isExternalThrone(backCoord)) {
+            if (this.config.KING_FAR_FROM_CENTRAL_THRONE_CAN_BE_SANDWICHED) {
+                return kingCoord;
+            }
```

Comment:
otherwise ?

Reply:
Otherwise not captured ! see after push, j'ai un peu enhancé !